cmd/root.go

```diff
@@ -45,18 +50,81 @@ var (
 	// SignalReceived is set to true when a signal causes us to exit. This makes
 	// determining the error message to show on exit much more easy.
 	SignalReceived = false
+
+	// profileCleanup is called after the root command is executed to
+	// cleanup a running cpu profile.
+	profileCleanup func()
 )
 
+// rootPreRun is executed before the root command runs and sets up cpu
+// profiling.
+//
+// Bassed on https://golang.org/pkg/runtime/pprof/#hdr-Profiling_a_Go_program
+func rootPreRun(*cobra.Command, []string) error {
+	if cpuProfile == "" {
+		return nil
+	}
+
+	f, err := os.Create(cpuProfile)
+	if err != nil {
+		return log.Printf("unable to create CPU profile file: %v", err)
```

Comment:
nit: this should be `fmt.Errorf("%w: unable to create CPU profile file", err)` instead of log.Printf?

Reply:
![image](https://user-images.githubusercontent.com/13023275/94852061-3020a180-03de-11eb-9e1d-47c58b70f3ec.png)

Reply:
No change is so small that one can't screw it up.... :roll_eyes:

Reply:
🤣